tests/operators/_asynchronous/test_bigquery.py

```diff
@@ -99,6 +99,8 @@ def test_dbt_run_airflow_async_bigquery_operator_execute(mock_build_and_run_cmd,
         dbt_kwargs={"task_id": "test_task"},
     )
 
+    operator.emit_datasets = False
```

Comment:
Do we have tests validating that the assets/datasets are being successfully emitted?

Reply:
We have integration tests where emit_datasets is True

Reply:
And those tests are passing in the main branch, which means they were not catching the problem we're fixing in this PR.

We should have tests that would reproduce the original problem, so we don't have regressions in future